### PR TITLE
fix(openclaw): restore peer_role routing

### PR DESCRIPTION
**Pull request by qin-ctx**
## Description

OpenClaw 插件的模块化重构丢失了 `peer_role` 对消息 `peer_id` 和数据面 actor 的分流逻辑。后续 session message 字段从 `role_id` 更名为 `peer_id` 后，回归直接表现为：配置 `peer_role=assistant` 时，user 消息仍写入发送者 `peer_id`，assistant 消息反而没有 `peer_id`。

本 PR 恢复三种模式的原始语义，并保持消息归属、session 生命周期和数据面 actor 三者分离：

- `assistant`：仅 assistant 消息写入 Agent peer；recall/search/read/import/delete 使用 Agent actor。
- `person`：仅 user 消息写入当前 sender peer；数据面使用当前 sender actor。
- `none`：消息不写 peer，数据面不发送 actor peer。
- session add/get/context/commit/compact、archive 和 tool-result 请求不携带 actor；commit 的记忆归属只由归档消息中的 `peer_id` 决定。

## Human Involvement

- [x] A human participated in the implementation or review loop
- [ ] This PR was generated entirely by AI agents without human participation in the loop

## Related Issue

暂无关联 Issue；问题由实际 OpenClaw 会话数据排查发现。相关历史 PR：

- [#2236 feat(memory): 引入 User/Peer 记忆隔离模型](https://github.com/volcengine/OpenViking/pull/2236)：引入 OpenClaw `peer_role` 的消息归属语义。
- [#2595 feat(plugins): use actor peer scope](https://github.com/volcengine/OpenViking/pull/2595)：明确 actor 只用于 recall/search/read/import/delete 等数据面请求，并从 session add/get/commit/compact 中移除。
- [#2654 refactor(openclaw): add setup and routing modules](https://github.com/volcengine/OpenViking/pull/2654)：模块化重构后，消息 peer 与数据面 actor 的角色分流丢失，并重新把 Agent actor 带入 session 生命周期，形成此次回归。
- [#3115 fix(openclaw): send peer_id for session messages](https://github.com/volcengine/OpenViking/pull/3115)：将 session message 字段从 `role_id` 更正为 `peer_id`，使潜在错误归因直接反映到服务端数据中。

## Type of Change

- [x] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Refactoring (no functional changes)
- [ ] Performance improvement
- [x] Test update

## Changes Made

- 恢复统一的 message peer 解析：只按 `peer_role + message.role` 决定消息体 `peer_id`。
- 恢复数据面 actor 路由：每次只读取当前 tool/hook context 的 sender，不使用 session sender 缓存；`person` 模式缺 sender 时直接失败。
- 清理 session 生命周期、archive 和 tool-result 中误传的 actor；commit 不再参与 peer 选择。
- 保持 trace 的 Agent 身份与请求 actor 分离；允许 `memory_forget` 识别 canonical peer memory URI。
- 删除本 PR 曾新增的一次性场景测试，仅更新主干已有测试中已经过时的 actor/peer 断言。

## Testing

- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [x] I have tested this on the following platforms:
  - [ ] Linux
  - [x] macOS
  - [ ] Windows

执行结果：

- `npm run typecheck`：通过。
- 除主干已知失败的 `architecture-boundaries.test.ts` 外：43 个现有测试文件、658 个现有测试全部通过，其中包含真实本地 HTTP 服务流程。
- 临时 peer-role 映射自测：1 个测试通过，验证后已删除，未提交到 PR。

## Checklist

- [x] My code follows the project's coding style
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published

## Screenshots (if applicable)

不适用。

## Additional Notes

当前 `main` 的 `architecture-boundaries.test.ts` 已存在 4 个与本次改动无关的失败，分别涉及 recall-trace 兼容导出、setup 直接 fetch、resource packager 边界和 memory URI 兼容导出，因此未勾选“全部现有单测通过”。本 PR 未修改这些违规来源。

### CODE REVIEW

**Review by LinQiang391 — APPROVED** [review]

File: examples/openclaw-plugin/services/context-lifecycle-service.ts [hunk 1]
```diff
@@ -926,7 +1014,7 @@ export async function afterTurnOpenVikingSession({
 
     const commitResult = await client.commitSession(ovSessionId, {
       wait: false,
-      agentId,
+      agentId: actorPeerId,
```
Comment (LinQiang391) [inline examples/openclaw-plugin/services/context-lifecycle-service.ts]: [P1] A person-mode session commit is scoped to the current or most recently cached sender actor. A group-chat archive batch can contain peer_ids for multiple people. The server derives the extraction scope from every peer_id in the batch, while the actor-peer view rejects reads and writes to other peer directories. Committing an Alice+Bob batch as Bob can therefore fail Phase 2 or omit Alice memories. Please avoid using one person actor for an aggregate session commit: split commits by peer, or use an internal commit identity constrained by the archived-message peer allowlist. Add a two-sender, one-session commit regression test.

File: examples/openclaw-plugin/plugin/openviking-memory-tools.ts [hunk 2]
```diff
@@ -216,7 +225,7 @@ export function registerOpenVikingMemoryTools(deps: OpenVikingMemoryToolsDeps):
               details: { action: "rejected", uri },
             };
           }
-          await client.deleteUri(uri, session.agentId);
+          await client.deleteUri(uri, session.actorPeerId);
```
Comment (LinQiang391) [inline examples/openclaw-plugin/plugin/openviking-memory-tools.ts:227]: [P1] memory_forget still cannot delete peer memories. The preceding isMemoryUri() check matches viking://user/{user}/memories/... but not the canonical viking://user/{user}/peers/{peer}/memories/... path. Exact-URI deletion is rejected, and query-based deletion filters every peer-memory candidate out. This leaves the assistant/person peer memories restored by this PR impossible to forget through the tool. Please extend the memory URI matcher and cover both exact-URI and search-then-delete paths.

File: examples/openclaw-plugin/plugin/openviking-session-routing-runtime.ts [hunk 3]
```diff
@@ -61,6 +70,28 @@ export function createOpenVikingSessionRoutingRuntime(options: {
...
+      .filter((value): value is string => Boolean(value));
+    const personPeerId = sanitizeOpenVikingPeerId(senderId);
+    if (personPeerId) {
+      for (const alias of aliases) {
+        sessionPersonPeerIds.set(alias, personPeerId);
+      }
+    }
+    return resolveOpenVikingActorPeerId({
+      peerRole,
+      personPeerId: personPeerId ?? aliases.map((alias) => sessionPersonPeerIds.get(alias)).find(Boolean),
```
Comment (LinQiang391) [inline examples/openclaw-plugin/plugin/openviking-session-routing-runtime.ts]: [P2] In person mode, when neither the current context nor the cache has a sender, this resolves to an undefined actor. ov_read, ov_list, imports, and other calls then omit the actor header, disabling the server peer-view filter; memory_store also silently falls back to self memory. Person mode should fail closed when no sender can be resolved instead of degrading to unscoped access. Please add a tool-context test with only sessionId and no requesterSenderId.

**Review by LinQiang391 — DISMISSED** [review]
The peer_role routing direction is sound, but two issues still block the core assistant/person behavior, plus one fail-open edge case in person mode. Please address these findings and add focused regression coverage before merging. Local verification: typecheck passed; 182 related tests passed across 6 test files. Existing tests do not cover multi-sender sessions, deletion of peer-memory URIs, or person mode without a sender.

**Review by LinQiang391 — COMMENTED** [review]
Non-blocking review suggestions: please consider the three inline findings around aggregate person-mode commits, peer-memory URI handling in memory_forget, and fail-closed behavior when a person sender cannot be resolved. These are recommendations for follow-up, not merge-blocking requirements.

File: examples/openclaw-plugin/services/context-lifecycle-service.ts (hunk 1 above)
Comment (qin-ctx) [reply examples/openclaw-plugin/services/context-lifecycle-service.ts]: Addressed in 59bdb8d4. I removed actor-peer routing from the whole session lifecycle (add/get/context/commit/compact/task polling), matching #2595. A commit no longer selects a peer or uses a cached sender; memory extraction remains driven by the peer_id values on archived messages. The sender cache was removed entirely. The focused check was run locally and not retained as a one-off PR test.

File: examples/openclaw-plugin/plugin/openviking-memory-tools.ts (hunk 2 above)
Comment (qin-ctx) [reply examples/openclaw-plugin/plugin/openviking-memory-tools.ts:227]: Addressed in 59bdb8d4. isMemoryUri now accepts the canonical viking://user/{user}/peers/{peer}/memories/... shape, so both exact-URI deletion and query-result filtering allow peer memories. Both paths were checked locally; the one-off scenario tests were not retained in the PR. The existing plugin suite passes (658 tests).

File: examples/openclaw-plugin/plugin/openviking-session-routing-runtime.ts (hunk 3 above)
Comment (qin-ctx) [reply examples/openclaw-plugin/plugin/openviking-session-routing-runtime.ts]: Addressed in 59bdb8d4. The session sender cache was removed. Person-mode actor routing now uses only the sender from the current tool/hook context, and resolveOpenVikingActorPeerId throws when that sender is missing, before any data request or memory write can fall back to an unscoped/self path. The focused check was run locally and not retained as a one-off PR test.

**Review by LinQiang391 — APPROVED** [review]
